### PR TITLE
attempt to stabilize tests

### PR DESCRIPTION
**Pull request by fosterseth**

### CODE REVIEW

File: pkg/controlsvc/controlsvc.go [hunk 1]
```diff
@@ -125,12 +125,14 @@ func (s *Server) AddControlFunc(name string, cType ControlCommandType) error {
 
 // RunControlSession runs the server protocol on the given connection.
 func (s *Server) RunControlSession(conn net.Conn) {
-	logger.Info("Client connected to control service\n")
+	logger.Info("Client connected to control service %s\n", conn.RemoteAddr().String())
```
Comment (eqrx) [inline pkg/controlsvc/controlsvc.go:128]: Could that just be 
`logger.Info("Client connected to control service %v\n", conn.RemoteAddr())`

File: pkg/controlsvc/controlsvc.go [hunk 2]
```diff
@@ -331,34 +335,37 @@ func (s *Server) RunControlSvc(ctx context.Context, service string, tlscfg *tls.
 					if ctx.Err() != nil {
 						return
 					}
+					if err != nil {
+						if strings.HasSuffix(err.Error(), "normal close") {
```
Comment (eqrx) [inline pkg/controlsvc/controlsvc.go:339]: where does this "normal close" come from? can this be done with the `errors.Is` method?

File: pkg/workceptor/command.go [hunk 3]
```diff
@@ -111,6 +112,13 @@ loop:
 			err = status.UpdateBasicStatus(statusFilename, WorkStateRunning, fmt.Sprintf("Running: PID %d", cmd.Process.Pid), stdoutSize(unitdir))
 			if err != nil {
 				logger.Error("Error updating status file %s: %s", statusFilename, err)
+				writeStatusFailures++
+				if writeStatusFailures > 3 {
+					logger.Error("Exceeded retries for updating status file %s: %s", statusFilename, err)
+					os.Exit(-1)
```
Comment (eqrx) [inline pkg/workceptor/command.go:118]: https://pkg.go.dev/os#Exit: "For portability, the status code should be in the range [0, 125].". Also os.Exit causes the process to exit immediately without executing any defers and such. Is that desired?

File: pkg/workceptor/remote_work.go [hunk 4]
```diff
@@ -302,15 +304,15 @@ func (rw *remoteUnit) monitorRemoteStatus(mw *utils.JobContext, forRelease bool)
 		_, err := conn.Write([]byte(fmt.Sprintf("work status %s\n", remoteUnitID)))
 		if err != nil {
 			logger.Debug("Write error sending to %s: %s\n", remoteUnitID, err)
-			_ = conn.Close()
+			_ = conn.(interface{ CloseConnection() error }).CloseConnection()
```
Comment (eqrx) [inline pkg/workceptor/remote_work.go:295]: Maybe make that a named interface instead of an anonymous one?

File: pkg/workceptor/remote_work.go [hunk 5]
```diff
@@ -302,15 +304,15 @@ func (rw *remoteUnit) monitorRemoteStatus(mw *utils.JobContext, forRelease bool)
...
 			conn = nil
 
 			continue
 		}
 		status, err := utils.ReadStringContext(mw, reader, '\n')
 		if err != nil {
 			logger.Debug("Read error reading from %s: %s\n", remoteNode, err)
-			_ = conn.Close()
+			_ = conn.(interface{ CloseConnection() error }).CloseConnection()
```
Comment (eqrx) [inline pkg/workceptor/remote_work.go:303]: See above

File: pkg/workceptor/workceptor.go [hunk 6]
```diff
@@ -480,57 +495,79 @@ func (w *Workceptor) GetResults(unitID string, startPos int64, doneChan chan str
 
 			break
 		}
-		var stdout *os.File
-		var err error
 		filePos := startPos
+		statChan := make(chan struct{}, 1)
```
Comment (eqrx) [inline pkg/workceptor/workceptor.go:499]: No need to buffer and send empty structs and sent empty structs into it. Just close the channel at the sender and the receiver will unblock

File: tests/functional/mesh/work_test.go [hunk 7]
```diff
@@ -23,9 +23,10 @@ func checkSkipKube(t *testing.T) {
 }
 
 func tearDown(controllers map[string]*receptorcontrol.ReceptorControl, m *mesh.CLIMesh) {
-	defer m.WaitForShutdown()
-	defer m.Destroy()
+	// defer m.WaitForShutdown()
```
Comment (eqrx) [inline tests/functional/mesh/work_test.go]: Why is that all commented out? Could you add a comment describing why this is done?

**Review by eqrx — CHANGES_REQUESTED** [review]

**Comment by fosterseth** [comment]
closing in favor of https://github.com/ansible/receptor/pull/487

**Pull request closed without merging**